### DISCUSSION TITLE
Tasmota Rules & Timers

### DISCUSSION THREAD

**Discussion by sag007** [arendst/Tasmota#20326] [Support request]
Hey Guys,

I have been searching through all the documents I can find for sometime now and still struggle to find the answers to my problems.

The scenario is that I have a freezer room that I would like to knock up the control gear for. Basically in normal operation using Tasmota I have a rule that cuts both the compressor and the internal circulation fan in and out and this takes place at between -19.0deg C to -21deg C. The rule I'm using for this process is below;

Rule1 
ON DS18B20#temperature>%var1% do backlog var1 999; var2 -19.0; power1 1; power2 1 ENDON ON DS18B20#temperature<%var2% do backlog var2 -999; var1 -21.0; Power1 0; Power2 0 ENDON

The problem I have is that once a day, when I'm home I need to disable Rule1, turn off both power 1 & 2 and then turn on power3 which is the defrost element for 30mins. Once that has complete, I then turn of power3 and enable Rule1 again and things should be back to a normal running state.

Timer1 {"Enable":1, "Mode":0,"Time":"17:00","Window":0,"Days":"SMTWTFS","Repeat":1,"Action":3}
Rule2 ON clock#Timer1=1 DO backlog rule1 0; Power1 0; Power2 0; Power3 1 ENDON

Timer2 {"Enable":1,"Mode":0,"Time":"19:30","Window":0,"Days":"SMTWTFS","Repeat":1,"Action":3}
Rule3 ON clock#Timer2=1 DO rule1 1 ENDON

Unfortunately this didn't work and I'm sure you probably see what's wrong even though I can not. I have even tried to just have the disabling of rule1 but it didn't work;

Timer1 {"Enable":1, "Mode":0,"Time":"17:00","Window":0,"Days":"SMTWTFS","Repeat":1,"Action":3}
Rule2 ON clock#Timer1=1 DO rule1 0 ENDON

Greatly appreciate any help you guys could offer.

Regards,

Nathan

**Comment by sfromis** [comment]
Did you make sure to have the Timers feature as such enabled? Checkmark at the top of the timers page, or the command `Timers`.

**Reply by sag007** [reply]
Hey Sfromis,

When I run the "timers" command I get the following;

20:13:16.280 MQT: stat/Freezer/RESULT = {"Timers":"ON",
"Timer1":{"Enable":1,"Mode":0,"Time":"20:10","Window":0,"Days":"1111111","Repeat":1,"Output":4,"Action":3},
"Timer2":{"Enable":1,"Mode":0,"Time":"20:12","Window":0,"Days":"1111111","Repeat":1,"Output":4,"Action":3},
"Timer3":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer4":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer5":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer6":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer7":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer8":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer9":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer10":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer11":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer12":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer13":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer14":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer15":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},
"Timer16":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0}}

Nathan
